packages/preview/src/browser/preview-handler.ts

```diff
@@ -31,15 +40,67 @@ export namespace RenderContentParams {
+ */
 export interface PreviewHandler {
+    /**
+     * One or more classes which shall be applied to the preview widget icon.
+     */
     readonly iconClass?: string;
+    /**
+     * Indicates whether and with which priority (larger is better) this preview handler is responsible for the given resource identified by the given URI.
+     * 
+     * @param uri the URI identifying a resource.
+     * 
+     * @returns A number larger than 0 if the handler is applicable, 0 or a negative number otherwise.
+     */
     canHandle(uri: URI): number;
+    /**
+     * Render the preview content by returning appropriate HTML.
+     * 
+     * @param params Information for the handler to render its content. Contains at least the textual content and the document URI.
+     * 
+     * @returns the HTMLElement which will be attached to the preview widget.
+     */
     renderContent(params: RenderContentParams): MaybePromise<HTMLElement | undefined>;
+    /**
+     * Search and return the HTMLElement which corresponds to the given fragment.
+     * This is used to initially reveal elements identified via the URI fragment.
+     * 
+     * @param content the preview widget element containing the content previously rendered by {@link PreviewHandler.renderContent}. 
+     * @param fragment the URI fragment for which the corresponding element shall be returned
+     * 
+     * @returns the HTMLElement which is part of content and corresponds to the given fragment, undefined otherwise.
+     */
     findElementForFragment?(content: HTMLElement, fragment: string): HTMLElement | undefined;
+    /**
+     * Search and return the HTMLElement which corresponds to the given line number.
+     * This is used to scroll the preview when the source editor scrolls.
+     * 
+     * @param content the preview widget element containing the previously rendered by {@link PreviewHandler.renderContent}. 
+     * @param sourceLine the line number for which the corresponding element shall be returned.
+     * 
+     * @returns the HTMLElement which is part of content and corresponds to the given line number, undefined otherwise.
+     */
     findElementForSourceLine?(content: HTMLElement, sourceLine: number): HTMLElement | undefined;
+    /**
+     * Returns the line number which corresponds to the preview element at the given offset.
+     * This is used to scroll the source editor when the preview scrolls.
+     * 
+     * @param content the preview widget element containing the previously rendered by {@link PreviewHandler.renderContent}. 
+     * @param offset the total amount by which the preview widget is scrolled.
+     * 
+     * @returns the source line number which corresponds to the preview element at the given offset, undefined otherwise.
+     */
     getSourceLineForOffset?(content: HTMLElement, offset: number): number | undefined;
 }
 
+/**
+ * Provider managing the available PreviewHandlers. 
+ */
```

Comment:
I don't think the following documentation provides any value.
We can either omit it or enhance it to describe the provider better.

Reply:
That's true there is not too much value in documenting this class. The most important point is that `findContributions` already sorts the returned list instead of only being a filter. For consistency I documented the rest of the class too. Would be fine with me to remove the documentation for it completely if you prefer that.

Reply:
I think it's fine to leave it, for consistency it is there even thought it might not provide much value that the name does not cover. It is always possible to improve documentation in future iterations.